codegen/lib/graphql_java_gen.rb

```diff
@@ -253,10 +253,15 @@ def java_implements(type)
     "implements #{interfaces.to_a.join(', ')} "
   end
 
-  def java_annotations(field)
-    @annotations.map do |annotation|
+  def java_annotations(field, in_argument: false)
+    annotations = @annotations.map do |annotation|
       "@#{annotation.name}" if annotation.annotate?(field)
-    end.compact.join("\n")
+    end.compact
+    if in_argument
+      annotations.join(" ") + " "
```

Comment:
The change I was thinking of was to just change 

```diff
-    end.compact.join("\n")
+    end.compact.join(separator)
```

where `separator` is an argument passed into `java_annotations`.  However, it might be even simpler to just have this method return an array and do the `.join("\n")` or `.join(' ')` on the return value in the template.